lighthouse-core/audits/injected-iframes.js

```diff
@@ -0,0 +1,200 @@
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+
+const Audit = require('./audit.js');
+const i18n = require('./../lib/i18n/i18n.js');
+const ComputedUserTimings = require('../computed/user-timings.js');
+const TraceProcessor = require('../lib/tracehouse/trace-processor.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides a potential cause of CLS. This descriptive title is shown to users when no iframe is injected in a time window before a LayoutShift event. */
+  title: 'Injected iframes likely didn\'t contribute to CLS',
+  /** Title of a Lighthouse audit that provides a potential cause of CLS. This descriptive title is shown to users when an iframe is injected in a time window before a LayoutShift event. */
+  failureTitle: 'Injected iframes potentially contributed to CLS',
+  /** Label for a column in a data table; entries will be the timestamp in milliseconds an event occurs. */
+  columnTimestamp: 'Timestamp',
+  /** Label for audit identifying the number of iframes likely affecting the page. */
+  displayValue: `{itemCount, plural,
+    =1 {1 iframe}
+    other {# iframes}
+    }`,
+  /** Description of a Lighthouse audit that tells the user potential causes of CLS. This is displayed after a user expands the section to see more. No character length limits. 'Learn More' becomes link text to additional documentation. */
+  description: 'Injecting an iframe with a correctly sized container can reduce layout shifting and improve CLS. [Learn More](https://web.dev/optimize-cls/#ads-embeds-and-iframes-without-dimensions)',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+/**
+ * LayoutShift timing windows are determined by trace events. We consider a timing window to be 
+ * between a ScheduleStyleRecalculation event and the end of its possibly corresponding UpdateLayerTree event.
+ * We assume that an UpdateLayerTree event belongs to a ScheduleStyleRecalculation event if it occurs within
+ * 20ms. This window is then determined to be a "layoutShiftWindow", if a LayoutShift event occurs
+ * within a time frame of this UpdateLayerTree event.
+ * @param {{ event: LH.TraceEvent; timing: any; duration: number; }[]} layoutEvents
+ * @return {Array<LH.Artifacts.DOMWindow>}
+ */
+function getLayoutShiftWindows(layoutEvents) {
+  /**@type {Array<LH.Artifacts.DOMWindow>} */
+  const shiftWindows = [];
+
+  // TODO: use a map for this ?
+  for (let i = 0; i < layoutEvents.length; i++) {
```

Comment:
an "exit early" coding style helps readability _alot_. check it out https://medium.com/swlh/return-early-pattern-3d18a41bba8

here, that'd mean using `continue` every time you have an `if` statement, instead of making a new scope